src/storage/cache.coffee

```diff
@@ -0,0 +1,49 @@
+_ = require('underscore')
+ApplicationState = require('./application_state')
+
+CACHE_KEY_PREFIX = 'cache:'
+
+class CacheItem
+  constructor: (@value, ttl) ->
+    @expiresAt = new Date(new Date().getTime() + (ttl * 1000))
+
+class ItemCache
+  constructor: (@cache, @key, @opts) ->
+
+  get: ->
+    @cache.get(@key)
+
+  set: (value) ->
+    @cache.set(@key, value, @opts)
+
+class Cache
+  getCacheFor: (key, opts) ->
```

Comment:
No, this is the same difference between ApplicationState and JsonSetting. 

the constructor method instead of just a `new ItemCache` is debatable, but my idea behind this: if the internal of the cache change this is hidden form the ItemCache users.